extensions/amp-story/1.0/test/test-amp-story-affiliate-link.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AFFILIATE_LINK_BUILT,
+  AmpStoryAffiliateLink,
+} from '../amp-story-affiliate-link';
+import {Services} from '../../../../src/services';
+
+describes.realWin('amp-story-affiliate-link', {amp: true}, env => {
+  let win;
+  let link;
+  const text = 'amp.devamp.devamp.devamp.devamp';
+
+  beforeEach(() => {
+    win = env.win;
+    const {sandbox} = env;
+    const resources = Services.resourcesForDoc(env.ampdoc);
+    sandbox.stub(resources, 'mutateElement').callsFake((element, mutator) => {
+      mutator();
+    });
+
+    const linkEl = win.document.createElement('a');
+    linkEl.setAttribute('affiliate-link-icon', 'shopping-cart');
+    linkEl.setAttribute('href', 'https://amp.dev');
+    linkEl.setAttribute('target', '_blank');
+    linkEl.insertAdjacentText('afterbegin', text);
+
+    const storyGridLayerEl = win.document.createElement('amp-story-grid-layer');
+    storyGridLayerEl.appendChild(linkEl);
+
+    const storyPageEl = win.document.createElement('amp-story-page');
+    storyPageEl.appendChild(storyGridLayerEl);
+
+    const storyEl = win.document.createElement('amp-story');
+    storyEl.appendChild(storyPageEl);
+
+    win.document.body.appendChild(storyEl);
+
+    link = new AmpStoryAffiliateLink(win, linkEl);
+  });
+
+  it('should build affiliate link', () => {
+    link.build();
+
+    expect(link.element_.hasAttribute('expanded')).to.be.false;
+    expect(link.element_.hasAttribute('pristine')).to.be.true;
+
+    expect(link.textEl_).to.not.be.null;
+    expect(link.textEl_.hasAttribute('hidden')).to.be.true;
+
+    expect(link.launchEl_).to.not.be.null;
+    expect(link.launchEl_.hasAttribute('hidden')).to.be.true;
+
+    expect(link.text_).to.equal(text);
+
+    expect(link.element_[AFFILIATE_LINK_BUILT]).to.be.true;
```

Comment:
Can we avoid referencing ivars here? E.g. use `linkEl` instead of `link.element_` and `linkEl.querySelector(...)` instead of `link.textEl_` etc.

Alternatively you can expose public methods e.g. `AmpStoryAffiliateLink.getTextElementForTesting()`.

Reply:
Done, thanks for the review.